src/cartridges/bm_adyen/cartridge/static/default/adyenSettings.js

```diff
@@ -33,30 +47,85 @@ document.addEventListener('DOMContentLoaded', () => {
     });
   }
 
-  // add event for save button availability on form change.
+  function hideAlertsOnTest() {
+    document.getElementById('settingsFormSubmitButton').click();
+    document.getElementById('saveChangesAlert').hide();
+    document.getElementById('notSavedChangesAlert').hide();
+    document.getElementById('testConnectionButton').disabled = true;
+  }
+
+  function showAlertsOnSave() {
+    document.getElementById('saveChangesAlert').show();
+    document.getElementById('notSavedChangesAlert').show();
+  }
+
+  // if browser is safari it sets custom padding
+  function checkBrowserSupport() {
+    if (/^((?!chrome|android).)*safari/i.test(navigator.userAgent)) {
+      formBody.style.setProperty('padding-top', '3rem');
+    }
+  }
+
+  testConnectionButton.addEventListener('click', hideAlertsOnTest);
+
   form.addEventListener('input', enableformButtons);
 
+  submitButton.addEventListener('click', showAlertsOnSave);
+
+  window.addEventListener('load', checkBrowserSupport);
+
   // add event listener to maintain form updates
   form.addEventListener('change', (event) => {
-    const { target } = event;
-    const { name } = target;
+    const { name } = event.target;
+    let { value } = event.target; // get checked boolean value for checkboxes
```

Comment:
```suggestion
    let { value } = event.target; // get checked boolean value for checkboxes
    if (event.target.type === 'checkbox') {
      value = event.target.checked;
    } // convert radio button strings to boolean if values are 'true' or 'false'
        if (event.target.type === 'radio') {
      if (event.target.value === 'true') {
        value = event.target.value;
      }

      if (event.target.value === 'false') {
        value = event.target.value;
      }
    }
```
let value = event.target.type === 'radio' ? event.target.value : event.target.checked;